config/prow/cluster/build_serviceaccounts.yaml

```diff
@@ -44,4 +44,12 @@ metadata:
   name: kubernetes-external-secrets-sa
   namespace: default
 ---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    iam.gke.io/gcp-service-account: k8s-csi-test@secretmanager-csi-build.iam.gserviceaccount.com
```

Comment:
Maybe a comment here would be good for reference: "#Used by the gcp provider for secrets-store-csi-driver"